### PR TITLE
Expose new feature flags

### PR DESCRIPTION
**Pull request by qnkhuat**
per [this](https://www.notion.so/metabase/Grandfather-accidentally-exposed-EE-features-for-existing-customers-d5d70d72f606420fab7b7e02c7b0855e) notion doc

This PR exposes a set of new feature flags: `:official_collections, :snippet_collections, :disable_password_login`

### CODE REVIEW

**Review by ranquild — DISMISSED** [review]

File: src/metabase/public_settings/premium_features.clj [hunk 1]
```diff
@@ -303,10 +307,17 @@
...
+  "Should we enable Question verifications (and more in the future, like workflows, forking, etc.)?"
   :content-management)
 
+(define-premium-feature ^{:added "0.41.0"} enable-official-collections?
+  "Should we enable Official Collections?"
+  :official-collections)
+
+(define-premium-feature ^{:added "0.41.0"} enable-snippet-folders?
+  "Should we enable Snippet folders?"
+  :snippet-folders)
```
Comment (ranquild) [inline src/metabase/public_settings/premium_features.clj:319]: `snippet-collections`?

**Review by ranquild — APPROVED** [review]

File: enterprise/backend/test/metabase_enterprise/public_settings_test.clj [hunk 2]
```diff
@@ -15,3 +17,30 @@
     (public-settings/enable-password-login! false)
     (is (= false
            (public-settings/enable-password-login)))))
+
+(deftest properties-token-features-test
```
Comment (calherries) [inline enterprise/backend/test/metabase_enterprise/public_settings_test.clj:21]: Not sure why this is test is in this namespace, given it's testing the API. 

I think it should go in `test/metabase/api/session_test.clj` near the other GET "/session/properties" tests https://github.com/metabase/metabase/blob/9515d70bceb3c600f23f9e5b4b8314028690b8bf/test/metabase/api/session_test.clj#L353

**Review by calherries — APPROVED** [review]
Approved because moving the test is super low priority and can be done later

File: enterprise/backend/test/metabase_enterprise/public_settings_test.clj (hunk 2 above)
Comment (noahmoss) [reply enterprise/backend/test/metabase_enterprise/public_settings_test.clj:21]: Agree...in general EE tests (and code) could use a reorganization

**Review by noahmoss — APPROVED** [review]

**Pull request merged**